PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseOfBlockOnAsyncClientsCheckTest.java

```diff
@@ -0,0 +1,175 @@
+     */
+    @Test
+    public void testUseOfBlockOnAsyncClientsVisitor() {
+
+        // assert visitor
+        assertVisitor();
+
+        int NUMBER_OF_INVOCATIONS = 1;
+        String METHOD_NAME = "block";
+        String AZURE_PACKAGE = "com.azure.";
+
+        // verify register problem
+        verifyRegisterProblem(METHOD_NAME, AZURE_PACKAGE, NUMBER_OF_INVOCATIONS);
+
+    }
+
+    /**
+     * This test method tests the use of blockOptional() method on async clients.
+     * This method should be flagged by the inspection tool.
+     */
+    @Test
+    public void testUseOfDifferentBlockOnAsyncClientsVisitor() {
+
+        // assert visitor
+        assertVisitor();
+
+        int NUMBER_OF_INVOCATIONS = 1;
+        String METHOD_NAME = "blockOptional";
+        String AZURE_PACKAGE = "com.azure.";
+
+        // verify register problem
+        verifyRegisterProblem(METHOD_NAME, AZURE_PACKAGE, NUMBER_OF_INVOCATIONS);
+    }
+
+    /**
+     * This test method tests the use of blockFirst() method on non-azure async clients.
+     * This method should not be flagged by the inspection tool.
+     */
+    @Test
+    public void testUseOfBlockOnAsyncClientsVisitorWithNonAzureClient() {
+
+        // assert visitor
+        assertVisitor();
+
+        int NUMBER_OF_INVOCATIONS = 0;
+        String METHOD_NAME = "blockFirst";
+        String AZURE_PACKAGE = "com.notAzure.";
+
+        // verify register problem
+        verifyRegisterProblem(METHOD_NAME, AZURE_PACKAGE, NUMBER_OF_INVOCATIONS);
+    }
+
+    /**
+     * This test method tests the use of a different method call on async clients.
+     * This method should not be flagged by the inspection tool.
+     */
+    @Test
+    public void testVisitOnDifferentMethodCall() {
+
+        // assert visitor
+        assertVisitor();
+
+        int NUMBER_OF_INVOCATIONS = 0;
+        String METHOD_NAME = "nonBlockingMethod";
+        String AZURE_PACKAGE = "com.azure.";
+
+        // verify register problem
+        verifyRegisterProblem(METHOD_NAME, AZURE_PACKAGE, NUMBER_OF_INVOCATIONS);
+    }
+
+    // Create a visitor object for the test
+    JavaElementVisitor createVisitor() {
+        boolean isOnTheFly = true;
+        UseOfBlockOnAsyncClientsVisitor visitor = new UseOfBlockOnAsyncClientsCheck.UseOfBlockOnAsyncClientsVisitor(mockHolder, isOnTheFly);
+        return visitor;
+    }
+
+    // Assert that the visitor object is not null and is an instance of JavaElementVisitor
+    void assertVisitor() {
+        assertNotNull(mockVisitor);
+        assertTrue(mockVisitor instanceof JavaElementVisitor);
+    }
+
+    // Verify that the registerProblem method is called the expected number of times
+    void verifyRegisterProblem(String METHOD_NAME, String AZURE_PACKAGE, int NUMBER_OF_INVOCATIONS) {
+
+        // Arrange
+        PsiReferenceExpression referenceExpression = mock(PsiReferenceExpression.class);
```

Comment:
Is it possible to compile an expression rather than mocking everything?

Reply:
use the code to create an instance of reference expression -- using a string input of source code

Reply:
opted to keep the current implementation